criu/pipes.c

```diff
@@ -463,18 +463,16 @@ int dump_one_pipe_data(struct pipe_data_dump *pd, int lfd, const struct fd_parms
 	if (pb_write_one(img, &pde, PB_PIPE_DATA))
 		goto err_close;
 
-	if (bytes) {
+	while (bytes > 0) {
 		int wrote;
-
 		wrote = splice(steal_pipe[0], NULL, img_raw_fd(img), NULL, bytes, 0);
 		if (wrote < 0) {
 			pr_perror("Can't push pipe data");
 			goto err_close;
```

Comment:
Maybe we also should add a check for `-EAGAIN` here and continue the loop, or at least retry several more times

Reply:
Hello Mike, I don't think `-EAGAIN` is useful to check here as splice() is running in blocking mode here, as we don't pass `SPLICE_F_NONBLOCK`